File: backend/app/models/trip.py
from pydantic import BaseModel, Field
from typing import List, Optional

class TripInDB(BaseModel):
    """Cấu trúc dữ liệu chuyến xe lưu trữ vật lý trong MongoDB"""
    trip_code: str = Field(..., description="Mã chuyến xe (VD: TRIP-2026-001)")
    driver_id: Optional[str] = Field(None, description="Mã _id của tài xế phụ trách")
    
    # Danh sách các mã đơn hàng trong chuyến này (giữ đúng thứ tự giao hàng)
    order_ids: List[str] = Field(default=[], description="Danh sách các _id đơn hàng")
    
    vehicle_info: Optional[dict] = None
    
    status: str = Field(
        default="PLANNED", 
        description="Trạng thái: PLANNED, DISPATCHED, IN_PROGRESS, COMPLETED, CANCELLED"
    )
    
    # Metadata hành trình
    total_distance_km: float = 0.0
    estimated_duration_min: int = 0
    
    actual_start_at: Optional[str] = None
    actual_end_at: Optional[str] = None
    
    created_at: str
    updated_at: str
    created_by: Optional[str] = None

File: backend/internal/trip/entity/trip.go
package entity

import "time"

type Trip struct {
	ID                   int64      `json:"id" db:"id"`
	TripCode             string     `json:"trip_code" db:"trip_code"`
	DriverID             *int64     `json:"driver_id" db:"driver_id"`
	VehicleLicensePlate  string     `json:"vehicle_license_plate" db:"vehicle_license_plate"`
	Status               string     `json:"status" db:"status"`
	TotalDistanceKm      float64    `json:"total_distance_km" db:"total_distance_km"`
	EstimatedDurationMin int        `json:"estimated_duration_min" db:"estimated_duration_min"`
	ActualStartAt        *time.Time `json:"actual_start_at" db:"actual_start_at"`
	ActualEndAt          *time.Time `json:"actual_end_at" db:"actual_end_at"`
	CreatedAt            time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt            time.Time  `json:"updated_at" db:"updated_at"`
	CreatedBy            string     `json:"created_by" db:"created_by"`
}
